.github/workflows/build_and_test.yml

```
@@ -0,0 +1,308 @@
+    name: Build and Test
+    runs-on: ubuntu-latest
+    env:
+      MIX_ENV: test
+      WANDERER_NOTIFIER_API_TOKEN: ${{ secrets.NOTIFIER_API_TOKEN }}
+      DISCORD_BOT_TOKEN: ${{ secrets.FAKE_DISCORD_TOKEN }}
+      WANDERER_DISCORD_BOT_TOKEN: ${{ secrets.FAKE_DISCORD_TOKEN }}
+      CACHE_DIR: "/tmp/wanderer_notifier_cache"
+      WANDERER_CACHE_DIR: "/tmp/wanderer_notifier_cache"
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
```

actionlint (1.7.4)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)